android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +80,102 @@ public void init(String siteKey, String touchpointUri, String apiKey,
   }
 
   @ReactMethod
-  public void sendInteraction(String interaction, ReadableMap propertiesMap, final Promise promise) {
+  public void sendInteraction(String interactionPath, ReadableMap propertiesMap, final Promise promise) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
     final OneRequest sendInteractionRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-    final OneCall sendInteractionCall = One.sendInteraction(sendInteractionRequest);
-
-    if (promise == null) {
-      sendInteractionCall.enqueue(null);
-    } else {
-      sendInteractionCall.enqueue(new OneCallback() {
-        @Override
-        public void onSuccess(@NotNull OneResponse oneResponse) {
-          notifyResult(promise,
-            "{\"captures: " + oneResponse.getCaptures().toString() + ", " +
-                  "\"interactionPath\": \"" + oneResponse.getInteractionPath().getValue().toString() + "\", " +
-                  "\"optimizations: " + oneResponse.getOptimizations().toString() + ", " +
-                  "\"statusCode\": " + oneResponse.getHttpStatusCode() + ", " +
-                  "\"tid\": \"" + oneResponse.getTid() + "\", " +
-                  "\"trackers\": " + oneResponse.getTrackers().toString() + "}"
-          );
-        }
-
-        @Override
-        public void onError(@NotNull OneSDKError oneSDKError) {
-          notifyProblem(promise, "" + oneSDKError.getSystemCode(), oneSDKError.getErrorMessage());
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
```

Comment:
suggestion, but not required:

make the common first arg of `throwErrors` a constant for better code reviews and readability without IDE support.

```
private static final boolean THROW_ERRORS = true;
//...
One.sendInteraction(THROW_ERRORS, sendInteractionRequest).join();
```

Reply:
Updated